src/Composer/Action/RequirePackage.php

```diff
@@ -69,6 +77,7 @@ public function execute(array $package)
         }
 
         // Get the Composer repos
+        // TODO What is repos used for?
         $repos = $composer->getRepositoryManager()->getRepositories();
```

Comment:
Looks like PlatformRepository is used for determining requirements like "ext-*", php, etc. But it is never getting add to the Pool given to the VersionSelector.